components/builder-core/src/api.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::result;
+use std::str::FromStr;
+
+use hab_net::routing::Broker;
+use protocol::originsrv::{CheckOriginAccessRequest, CheckOriginAccessResponse, Origin,
+                          OriginChannel, OriginChannelCreate, OriginChannelGet, OriginGet,
+                          OriginPackage, OriginPackageGet, OriginPackageGroupPromote,
+                          OriginPackageIdent, OriginPackagePromote};
+use protocol::net::{ErrCode, NetError, NetOk};
+use protocol::scheduler::{Group, GroupGet, GroupState, Project, ProjectState};
+use protocol::sessionsrv::{Session, SessionCreate, SessionGet};
+
+use data_structures::UnsuccessfulJobGroupPromote;
+
+use error::{Error, Result};
+
+pub fn get_origin<T: ToString>(origin: T) -> result::Result<Option<Origin>, NetError> {
+    let mut conn = Broker::connect().unwrap();
```

Comment:
I tried to crack this nut about 20 different ways, and all of them either failed or felt terrible.  In the end, I chatted with @reset about how we were testing this stuff and decided that writing integration tests for it, using our existing integration test setup, was a better way to go than trying different ways of unit testing this.  Those integration tests are not part of this PR, as some of them proved very challenging to get working, and I didn't want this PR to wilt on the vine.